cpp/util/Histogram.h

```diff
@@ -72,13 +72,13 @@ class Axis
     virtual size_t bin(const float& value) const = 0;
 
     //! Return the boundaries of bins.
-    std::vector<float> getBinEdges() const
+    const std::vector<float> getBinEdges() const
```

Comment:
This comment goes for every new `const` return type declaration in this file: Either return the `std::vector` and allow a copy to be made or return by `const std::vector<>&`

Reply:
Would there be any preference in one over the other in this case?

Reply:
The decision would have to be made on a case-by-case basis. Anytime you are returning a local variable you should return by `std::vector<>`, most other cases it would probably be better to return by `const std::vector<>&`. If you want more advice with this we should talk in person.

Reply:
The difference has to do with ownership and mutability. If the object is returning one of its members, it should return by constant reference (`const std::vector<T>&`) to avoid returning a new copy of the data, while also ensuring that the returned reference is immutable (constant) so that the calling code cannot mutate this object's internal state. If the vector being returned is not a member (for example, if it is allocated and filled by the function being called) then you must return `std::vector<T>` -- note that this cannot be constant because it is a returning by copy (or move if the copy is elided), and it cannot be a reference because the data's lifetime is limited to the function body and the reference would be invalid.

Reply:
Reading this again, I am not sure what encouraged me to change this file at all. I think I was trying to fix warnings from VSCode but I'm not sure if this change actually does anything. Maybe let's just revert the changes in this file, or perhaps only make changes that return class members by constant reference (where applicable) and revert the rest.

Reply:
I tried to do the latter suggestion from @bdice last comment. Let me know if I was successful.